core/src/main/java/tc/oc/pgm/structure/StructureModule.java

```diff
@@ -0,0 +1,115 @@
+package tc.oc.pgm.structure;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Logger;
+import org.bukkit.util.Vector;
+import org.jdom2.Attribute;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.MapProtos;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.filters.FilterParser;
+import tc.oc.pgm.filters.StaticFilter;
+import tc.oc.pgm.filters.dynamic.FilterMatchModule;
+import tc.oc.pgm.regions.CuboidValidation;
+import tc.oc.pgm.snapshot.SnapshotMatchModule;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.Node;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class StructureModule implements MapModule<StructureMatchModule> {
+
+  private final List<DynamicDefinition> dynamics;
+
+  public StructureModule(List<DynamicDefinition> dynamics) {
+    this.dynamics = dynamics;
+  }
+
+  @Nullable
+  @Override
+  public StructureMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new StructureMatchModule(match, this.dynamics);
+  }
+
+  @Nullable
+  @Override
+  public Collection<Class<? extends MatchModule>> getHardDependencies() {
+    return ImmutableList.of(FilterMatchModule.class, SnapshotMatchModule.class);
+  }
+
+  public static class Factory implements MapModuleFactory<StructureModule> {
+
+    @Nullable
+    @Override
+    public StructureModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      if (factory.getProto().isOlderThan(MapProtos.FILTER_FEATURES)) return null; // TODO
+
+      final Map<String, StructureDefinition> structures = new HashMap<>();
+      for (Element elStruct :
+          XMLUtils.flattenElements(doc.getRootElement(), "structures", "structure")) {
+        final StructureDefinition definition =
+            new StructureDefinition(
+                elStruct.getAttribute("id").getValue(), // TODO Default?
```

Comment:
If someone doesn't set this, this is a NPE.

`XMLUtils.getRequiredAttribute("id").getValue()` if you want it required, otherwise, handle it somehow